engine/platform/src/platform_window.h

```diff
@@ -140,23 +162,42 @@ namespace dmPlatform
     uint32_t       GetWindowHeight(HWindow window);
     uint32_t       GetWindowStateParam(HWindow window, WindowState state);
     float          GetDisplayScaleFactor(HWindow window);
+    uintptr_t      GetProcAddress(HWindow window, const char* proc_name);
 
     int32_t        GetKey(HWindow window, int32_t code);
     int32_t        GetMouseButton(HWindow window, int32_t button);
     int32_t        GetMouseWheel(HWindow window);
     void           GetMousePosition(HWindow window, int32_t* x, int32_t* y);
+    uint32_t       GetTouchData(HWindow window, TouchData* touch_data, uint32_t touch_data_count);
+    bool           GetAcceleration(HWindow window, float* x, float* y, float* z);
+    const char*    GetJoystickDeviceName(HWindow window, uint32_t joystick_index);
+    uint32_t       GetJoystickAxes(HWindow window, uint32_t joystick_index, float* values, uint32_t values_capacity);
+    uint32_t       GetJoystickHats(HWindow window, uint32_t joystick_index, uint8_t* values, uint32_t values_capacity);
+    uint32_t       GetJoystickButtons(HWindow window, uint32_t joystick_index, uint8_t* values, uint32_t values_capacity);
 
     void           SetDeviceState(HWindow window, DeviceState state, bool op1);
     void           SetDeviceState(HWindow window, DeviceState state, bool op1, bool op2);
     bool           GetDeviceState(HWindow window, DeviceState state);
+    bool           GetDeviceState(HWindow window, DeviceState state, int32_t op1);
 
     void           SetWindowSize(HWindow window, uint32_t width, uint32_t height);
     void           SetSwapInterval(HWindow window, uint32_t swap_interval);
     void           SetKeyboardCharCallback(HWindow window, WindowAddKeyboardCharCallback cb, void* user_data);
     void           SetKeyboardMarkedTextCallback(HWindow window, WindowSetMarkedTextCallback cb, void* user_data);
     void           SetKeyboardDeviceChangedCallback(HWindow window, WindowDeviceChangedCallback cb, void* user_data);
+    void           SetGamepadEventCallback(HWindow window, WindowGamepadEventCallback cb, void* user_data);
+
     void           IconifyWindow(HWindow window);
     void           PollEvents(HWindow window);
+    void           SwapBuffers(HWindow window);
+
+    void*          AcquireAuxContext(HWindow window);
+    void           UnacquireAuxContext(HWindow window, void* aux_context);
+
+    // Platform specifics
+    int32_t        AndroidVerifySurface(HWindow window);
+    void           AndroidBeginFrame(HWindow window);
+    void           iOSSetViewTypeOpenGL(HWindow window);
```

Comment:
Another option is to put the platform specific things in a platform specific header, and only include that. E.g.:
```
#if defined(ANDROID)
    #include <platform_android.h>
#endif
```

etc, just like you did for opengl etc